Common/Source/Draw/LKDrawWaypoints.cpp

```diff
@@ -441,18 +457,26 @@ void MapWindow::DrawWaypointsNew(LKSurface& Surface, const RECT& rc, const Scree
 	      case DISPLAYNONE:
 		dowrite = (GetMultimap_Labels()<MAPLABELS_ONLYTOPO) || intask || islandable;
 		if (draw_alt) {
-		  if ( ArrivalValue == (ArrivalValue_t) avAltitude ) {
-			if ( (MapBox == (MapBox_t)mbUnboxedNoUnit) || (MapBox == (MapBox_t)mbBoxedNoUnit) )
-			  _stprintf(Buffer, TEXT("%d"),
+		  switch(ArrivalValue)
+          {
+		    case  avAltitude:		    
+			  if ( (MapBox == (MapBox_t)mbUnboxedNoUnit) || (MapBox == (MapBox_t)mbBoxedNoUnit) )
+			    _stprintf(Buffer, TEXT("%d"),
 				   (int)(WayPointList[i].AltArivalAGL*ALTITUDEMODIFY));
-			else
-			  _stprintf(Buffer, TEXT("%d%s"),
+			  else
+			    _stprintf(Buffer, TEXT("%d%s"),
 				   (int)(WayPointList[i].AltArivalAGL*ALTITUDEMODIFY),
 				   sAltUnit);
-		  } else
+		    break;
+
+		    case avGR:
 			  _stprintf(Buffer, TEXT("%d"),
 				   (int)(WayPointCalc[i].GR) );
+		    break;
 
+		    default: _stprintf(Buffer, TEXT(""));
```

Comment:
empty string using sprintf are probably the slower method ...
"_tcscpy(Buffer, TEXT(""));" is a faster method
"Buffer[0] = '\0';" is fastest method ...

Reply:
agree, it's not critical now, but using sprintf for copy string is bad usecase ...